mm2src/coins/eth.rs

```diff
@@ -182,39 +199,27 @@ lazy_static! {
 
 pub type Web3RpcFut<T> = Box<dyn Future<Item = T, Error = MmError<Web3RpcError>> + Send>;
 pub type Web3RpcResult<T> = Result<T, MmError<Web3RpcError>>;
-pub type GasStationResult = Result<GasStationData, MmError<GasStationReqErr>>;
 type EthPrivKeyPolicy = PrivKeyPolicy<KeyPair>;
-type GasDetails = (U256, U256);
 
-#[derive(Debug, Display)]
-pub enum GasStationReqErr {
-    #[display(fmt = "Transport '{}' error: {}", uri, error)]
-    Transport {
-        uri: String,
-        error: String,
-    },
-    #[display(fmt = "Invalid response: {}", _0)]
-    InvalidResponse(String),
-    Internal(String),
+#[derive(Clone, Debug)]
+pub(crate) struct LegacyGasPrice {
+    pub gas_price: U256,
```

Comment:
If `struct LegacyGasPrice` has `pub(crate)` visibility, its fields should have the same visibility level or more restrictive (like private). Please use `pub(crate) gas_price: U256,`

Reply:
I thought maybe pub for members is okay here, like `pub(crate)` for the struct level defines visibility for the whole struct and `pub` for members simply to open/close access to specific member

Reply:
> I thought maybe pub for members is okay here, like `pub(crate)` for the struct level defines visibility for the whole struct and `pub` for members simply to open/close access to specific member

:D reminds me of some of my favorite threads in the CLI or electrum connections PR. I will try to find it, as there a lot of review notes.

my answer is this: its about consistency of visibility. the visibility of struct fields should generally be **as restrictive as**, or more restrictive than, the **struct's own** visibility to maintain consistency and encapsulation. While `pub` on fields within a `pub(crate) struct` doesnt expose them outside of the crate due to the struct's limited visibility, using `pub(crate)` for fields explicitly aligns with the struct's intended encapsulation boundary. Also You shouldnt trust compiler and allow fields to have higher level of visibility then structure.

Reply:
@dimxy Here we go https://github.com/KomodoPlatform/komodo-defi-framework/pull/1768#discussion_r1222889458